app/src/main/java/io/apicurio/registry/rest/RegistryExceptionMapper.java

```diff
@@ -42,22 +44,24 @@ private static Error toError(Throwable t, int code) {
     
     private static Response toResponse(Throwable t, int code) {
         Error error = toError(t, code);
-        Response response = Response.status(code).type(MediaType.APPLICATION_JSON).entity(error).build();
-        return response;
+        return Response.status(code).type(MediaType.APPLICATION_JSON).entity(error).build();
     }
 
     /**
      * @see javax.ws.rs.ext.ExceptionMapper#toResponse(java.lang.Throwable)
      */
     @Override
-    public Response toResponse(StorageException exception) {
+    public Response toResponse(RegistryException exception) {
         int code = 500;
         if (exception instanceof AlreadyExistsException) {
             code = 409;
         }
         if (exception instanceof NotFoundException) {
             code = 404;
         }
+        if (exception instanceof RulesException) {
+            code = 403; // forbidden ?
```

Comment:
Probably 400 for "Bad Request".  Forbidden is reserved for auth I believe.

Reply:
OK, used 400 + used HttpURLConnection for the codes.